File: MeetUp.Domain/Models/Entities/Notification.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MeetUp.Domain.Models.Entities
{
    public class Notification
    {
        public int Id { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public string FromUserId { get; set; }
        [NotMapped]
        public virtual AppUser FromUser { get; set; }
        public string ToUserId { get; set; }
        [NotMapped]
        public virtual AppUser ToUser { get; set; }
        public int? PostId { get; set; }
        [NotMapped]
        public virtual Post Post { get; set; }
        public int NotificationTypeId { get; set; }
        public virtual NotificationType NotificationType { get; set; }
    }
}

File: client-app/src/app/models/notification.ts
import { User } from "./user";
import { NotificationType } from "./NotificationType";
import { Post } from "./post";
export interface Notification {
    id: number;
    createdDate: Date,
    fromUser: User
    fromUserId: string
    toUser: User
    toUserId: string
    post: Post
    postId: number
    notificationType: NotificationType
    notificationTypeId: number
}
export interface NotificationDto {
    id: number;
    createdDate: Date,
    fromUserName: string
    fromUserImage: string
    notificationTypeName: string
}